### ISSUE TITLE
[IntelliJ] Find error in idea.log when Log in with device code

### ISSUE THREAD

**Issue by t-rufang** [Microsoft/azure-tools-for-java#2368]
1. Put following string into `C:\Users\<username>\AzureToolsForIntelliJ\AadProvider.json`
```
{
    "EnableDeviceLogin": true
}
```
2. Sign in with device code
3. Check the idea.log, we can find the following error log
```
2018-11-20 15:31:26,082 [ 112271]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: 86c96d1a-5e11-4ed1-abef-585504db0600
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:25Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-11-20 15:31:31,493 [ 117682]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: 86c96d1a-5e11-4ed1-abef-585534db0600
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:31Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-11-20 15:31:36,902 [ 123091]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: c533a1b1-6ee3-4007-a864-ff2fd9120700
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:36Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-11-20 15:31:42,305 [ 128494]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: 4349aba8-6cf6-47b6-8438-e8cf15040700
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:42Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-11-20 15:31:47,631 [ 133820]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: 566d1139-5784-411a-b6b2-395272010700
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:47Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2018-11-20 15:31:53,044 [ 139233]  ERROR - d.adal4j.AuthenticationContext - [Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2] Execution of class com.microsoft.aad.adal4j.AcquireTokenCallable failed. 
com.microsoft.aad.adal4j.AuthenticationException: AADSTS70016: Pending end-user authorization.
Trace ID: 92d1a1f4-5759-4306-9015-8ec35af80600
Correlation ID: f7233c5b-b241-493e-a720-8bb51f9f5ef2
Timestamp: 2018-11-20 07:31:52Z
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:117)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by SummerSun** [comment]
The exception is thrown when pending for users to login with the code. This kind of exception should be swallowed. We catch the exception when geting result from the FutureTask, but not the exception when execute it.

The loop logic to acquire token by device code is handled by `ApplicationManager.getApplication().executeOnPooledThread`. The root cause is that the `AdalCallable` will just throw the exception if the callback is null. I will fix it by checking the callback and ensuring that we pass in a callback with the `onFailure` defined.

**Comment by Flanker32** [comment]
Fixed in pr #2690
